narwhals/_polars/group_by.py

```diff
@@ -19,23 +20,36 @@
 
 class PolarsGroupBy:
     _compliant_frame: PolarsDataFrame
-    _keys: Sequence[str]
+    _keys: Sequence[IntoExpr]
 
     @property
     def compliant(self) -> PolarsDataFrame:
         return self._compliant_frame
 
     def __init__(
-        self, df: PolarsDataFrame, keys: Sequence[str], /, *, drop_null_keys: bool
+        self,
+        compliant_frame: PolarsDataFrame,
+        keys: Sequence[PolarsExpr],
+        /,
+        *,
+        drop_null_keys: bool,
     ) -> None:
-        self._compliant_frame = df
-        self._keys = list(keys)
-        df = df.drop_nulls(keys) if drop_null_keys else df
-        self._grouped: NativeGroupBy = df._native_frame.group_by(keys)
+        if not drop_null_keys:
+            self._compliant_frame = compliant_frame
+            self._keys = [extract_native(arg) for arg in keys]
+        else:
+            compliant_frame = compliant_frame.with_columns(*keys)
+            self._keys = [extract_native(arg).meta.output_name() for arg in keys]
+            self._compliant_frame = compliant_frame.drop_nulls(self._keys)
+
+        self._grouped: NativeGroupBy = self._compliant_frame._native_frame.group_by(
+            self._keys
+        )
```

Comment:
So polars ended up being the ugliest, since there is not group_by param to deal with dropping nulls, but we support this (introduced for plotly)